### PR TITLE
Use xdg user dirs

### PR DESCRIPTION
**Pull request by wvffle**
I've added https://github.com/adrg/xdg to be able to get xdg dirs easily regardless of the user os.

Download dir now defaults to `XDG_DOWNLOAD_DIR` (#157)
Also added `GOMUKS_DOWNLOAD_DIR` env var to be able to specify the download dir manually.

### CODE REVIEW

**Comment by tulir** [comment]
github.com/adrg/xdg doesn't seem to actually read `user-dirs.dirs` (https://github.com/tulir/gomuks/issues/157#issuecomment-622473068)

**Comment by wvffle** [comment]
I just thought that I can load the user-dirs file with https://github.com/joho/godotenv and then call `xdg.Reload()` to fetch variables from env.

But I've noticed one thing that's actually making it pointless. System env variables will not be expanded. There are open PRs for that though: https://github.com/joho/godotenv/pull/103, https://github.com/joho/godotenv/pull/72. Without them `XDG_DOWNLOAD_DIR="$HOME/Downloads/gomuks"` will point to `/Downloads/gomuks`

**Comment by wvffle** [comment]
I've added some basic env file loader. it just reads and trims each line of passed file. If `line[0] == '#'` then it skips that line.
Then I split on first `=`, set the first word on left as a key and parse the value.

Value can be in `'` or `"` or without them, it considers end of the value at next unescaped occurrence of `'` or `"` respectively. (or at first unescaped ` ` if value is not in quotes) Rest of the line is skipped.

Variables in values are expanded, same goes with `~/` at the beginning of a value

@tulir, please check it as those are my first steps with golang :sweat_smile:

**Comment by normen** [comment]
go-ini has nice support for expanding env variables in case that helps: https://github.com/go-ini/ini

File: lib/util/envfile.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+			continue
+		}
+
+		escapeNext = false
+	}
+
+	value = builder.String()
+
+	// Expand ~ to home directory
+	if strings.HasPrefix(value, "~/") {
```
Comment (mhmdanas) [inline lib/util/envfile.go:87]: This would not expand a path like `~foo/bar`, which refers to the directory `bar` in the home directory of the user `foo`.

**Pull request closed without merging**